### PR TITLE
1) Throw obsolete warning message for Test-AzureResource and Test-Azu…

### PR DESCRIPTION
**Pull request by vivsriaus**
…reResourceGroup cmdlets.

2) Rename Test-AzureResourceGroupTemplate to Test-AzureResourceGroupDeployment
3) Remove deployment parameters from New-AzureResourceGroup
4) Update tests for all of the above

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vivsriaus__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (visriniv). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManager/ResourceManager/Commands.ResourceManager/Cmdlets/Implementation/TestAzureResourceCmdlet.cs [hunk 1]
```diff
@@ -131,6 +131,7 @@ protected override void OnProcessRecord()
         /// </summary>
         private void RunCmdlet()
         {
+            this.WriteWarning("This cmdlet is being deprecated. ");
```
Comment (ravbhatnagar) [inline src/ResourceManager/ResourceManager/Commands.ResourceManager/Cmdlets/Implementation/TestAzureResourceCmdlet.cs]: IMO we should change this to follow the standard PS cmdlets deprecation message as follows - "The Test-AzureResource cmdlet is deprecated and will be removed in a future release.”

File: src/ResourceManager/ResourceManager/Commands.ResourceManager/Cmdlets/Implementation/TestAzureResourceGroupCmdlet.cs [hunk 2]
```diff
@@ -57,6 +57,7 @@ protected override void OnProcessRecord()
         /// </summary>
         private void RunCmdlet()
         {
+            this.WriteWarning("This cmdlet is being deprecated. ");
```
Comment (ravbhatnagar) [inline src/ResourceManager/ResourceManager/Commands.ResourceManager/Cmdlets/Implementation/TestAzureResourceGroupCmdlet.cs]: IMO we should change this to follow the standard PS cmdlets deprecation message as follows - "The Test-AzureResourceGroup cmdlet is deprecated and will be removed in a future release.”

File: src/ResourceManager/ResourceManager/Commands.ResourceManager/Cmdlets/Implementation/TestAzureResourceCmdlet.cs (hunk 1 above)
Comment (vivsriaus) [inline src/ResourceManager/ResourceManager/Commands.ResourceManager/Cmdlets/Implementation/TestAzureResourceCmdlet.cs]: @ravbhatnagar Thanks for the feedback. Committed the changes

File: src/ResourceManager/ResourceManager/Commands.ResourceManager/Cmdlets/Implementation/TestAzureResourceGroupCmdlet.cs (hunk 2 above)
Comment (vivsriaus) [inline src/ResourceManager/ResourceManager/Commands.ResourceManager/Cmdlets/Implementation/TestAzureResourceGroupCmdlet.cs]: Done.

**Comment by markcowl** [comment]
@vivsriaus Why are you adding an extra step for creating a resource group and deployign a template - right now this is a single step and you are makign it two steps.  Why are you making this more difficult for customers?

**Comment by vivsriaus** [comment]
@markcowl Currently, there are two ways to submit deployments using powershell - one, using New-AzureResourceGroupDeployment and another using New-AzureResourceGroup. We want users to use New-AzureResourceGroupDeployment to submit deployments, and use New-AzureResourceGroup to create just the resource groups (hence matching our APIs). This is the experience we want for users going forward.

**Comment by vivsriaus** [comment]
@markcowl Spoke to Charles. We want this change to be included in the next POSH release (when we deprecate Switch-AzureMode). For now, I'll close this PR, and send out another PR that has a warning message for New-AzureResourceGroup (indicating the deployment parameters are being deprecated)

**Pull request closed without merging**